lib/cassandra-schema/migrator.rb

```diff
@@ -84,6 +105,32 @@ def update_version(target)
       @current_version = target
     end
 
+    def lock_schema
+      result = @connection.execute <<~CQL
+        INSERT INTO schema_information(name, value)
+        VALUES('lock', '1')
+        IF NOT EXISTS
+        USING TTL #{@options.fetch(:lock_timeout)}
+      CQL
+
+      result.rows.first.fetch("[applied]")
+    end
+
+    def renew_lock
+      @connection.execute <<~CQL
+        UPDATE schema_information
+        USING TTL #{@options.fetch(:lock_timeout)}
+        SET value = '1'
+        WHERE name = 'lock'
+      CQL
+    end
+
+    def unlock_schema
+      @connection.execute <<~CQL
+        DELETE FROM schema_information WHERE name = 'lock';
```

Comment:
Should we consider using `IF EXISTS` here?

Reply:
I'm not sure about this one. What would be the purpose of using `IF EXISTS` for this operation?

Reply:
I'll have to read up on the semantics again to be sure, but I think that this as written, runs the risk of writing a tombstone that will shadow a future write (called a "doomstone" by some folks).

Maybe we can avoid that risk by using a quorum-or-higher consistency level here (and on the other lock-related calls)? Maybe we should be doing that anyway?

Reply:
I agree with using  a quorum-or-higher consistency level  for operations related to `schema_information` queries. Updated it on https://github.com/tarolandia/cassandra-schema/pull/3/commits/6cfc43962cbcf4882b492845f9cbf646e3e9f82c